imap_processing/glows/l0/glows_decom.py

```diff
@@ -0,0 +1,59 @@
+from pathlib import Path
+
+from bitstring import ReadError
+from space_packet_parser import parser, xtcedef
+
+from imap_processing import imap_module_directory
+from imap_processing.glows.l0.glows_l0_data import GlowsDeL0, GlowsHistL0
+
+
+def decom_packets(packet_file: str) -> tuple[list[GlowsHistL0], list[GlowsDeL0]]:
+    """Decom GLOWS data packets using GLOWS packet definition
+    Parameters
+    ----------
+    packet_file : str
+        Path to data packet path with filename
+
+    Returns
+    -------
+    data : tuple[list[GlowsHistL0], list[GlowsDeL0]]
+        A tuple with two pieces: one list of the GLOWS histogram data, in GlowsHistL0
+        instances, and one list of the GLOWS direct event data, in GlowsDeL0 instance
+    """
+    hist_apid = 1480
+    de_apid = 1481
+
+    # Define paths
+    xtce_document = Path(
+        f"{imap_module_directory}/glows/packet_definitions/GLX_COMBINED.xml"
+    )
+
+    hist_packet_definition = xtcedef.XtcePacketDefinition(xtce_document)
+    histparser = parser.PacketParser(hist_packet_definition)
+
+    histdata = []
+    dedata = []
+
+    with open(packet_file, "rb") as binary_data:
+        try:
+            hist_packets = histparser.generator(
+                binary_data,
+                parse_bad_pkts=False,
+                buffer_read_size_bytes=5790778,
+            )
+
+            for packet in hist_packets:
+                # Do something with the packet data
+                if packet.header["PKT_APID"].derived_value == hist_apid:
+                    hist_l0 = GlowsHistL0(packet)
+                    histdata.append(hist_l0)
+
+                if packet.header["PKT_APID"].derived_value == de_apid:
+                    de_l0 = GlowsDeL0(packet)
+                    dedata.append(de_l0)
+
+        except ReadError as e:
```

Comment:
This is similar to the safe_packet_generator that Laura created in her Ultra L0 PR: #128. It does occur with the smaller packet that I created, because I ended up cutting it off in the middle of the last packet.

Reply:
> It does occur with the smaller packet that I created, because I ended up cutting it off in the middle of the last packet

I think this was covering up a true error in Ultra, so I'd be hesitant to put this in here. Are you able to trim the packet to the exact binary length you want? Maybe also put the command you did to trim the file in a comment at the top of this file or somewhere so that we can remember how to recreate the data here if we get a new test data dump.

Reply:
Yeah, I'll take a look at the packet and see if I can actually get a properly trimmed packet. I was using the code that the GLOWS team provided, which does byte by byte reading, but apparently failed to actually chop it at the right place. I'll try again, just wanted to get this PR up mostly

Reply:
Agree with Greg. Glad he caught that for Ultra.